File: aquila-frontend/src/lib/api/attribution.ts
import { AttributionResult, VesselCandidate, OriginEstimate, DriftResult } from './types';
import { apiClient } from './client';

export const attributionApi = {
  async evaluateCandidates(
    investigationId: string,
    origin: OriginEstimate,
    drift: DriftResult,
    candidates: VesselCandidate[]
  ): Promise<AttributionResult> {
    return apiClient.post<AttributionResult>('/attribution/evaluate', {
      investigation_id: investigationId,
      origin_estimate: origin,
      drift_result: drift,
      candidates: candidates
    });
  }
};

File: backend/app/api/v1/attribution.py
from fastapi import APIRouter

from app.schemas.attribution import AttributionQuery, AttributionResult
from app.services.attribution_service import AttributionService

router = APIRouter()
service = AttributionService()

@router.post("/evaluate", response_model=AttributionResult)
async def evaluate_attribution(query: AttributionQuery):
    """
    Evaluates AIS candidates against evidence factors.
    """
    return service.evaluate(
        investigation_id=query.investigation_id,
        origin=query.origin_estimate,
        drift=query.drift_result,
        candidates=query.candidates
    )
